core/src/test/scala/org/broadinstitute/dsde/rawls/StatsDTestUtils.scala

```diff
@@ -0,0 +1,62 @@
+package org.broadinstitute.dsde.rawls
+
+import java.util.concurrent.TimeUnit
+
+import com.codahale.metrics.SharedMetricRegistries
+import com.codahale.metrics.health.SharedHealthCheckRegistries
+import com.readytalk.metrics.{StatsD, StatsDReporter}
+import org.broadinstitute.dsde.rawls.model.SubmissionStatuses.SubmissionStatus
+import org.broadinstitute.dsde.rawls.model.WorkflowStatuses.WorkflowStatus
+import org.broadinstitute.dsde.rawls.model.{Submission, Workspace, WorkspaceName}
+import org.mockito.Mockito.{atLeastOnce, inOrder => mockitoInOrder}
+import org.scalatest.concurrent.Eventually
+import org.scalatest.mock.MockitoSugar
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+/**
+  * Created by rtitle on 6/29/17.
+  */
+trait StatsDTestUtils { this: MockitoSugar with Eventually with RawlsTestUtils =>
+
+  protected def withStatsD[T](testCode: => T)(verify: Seq[(String, String)] => Unit = _ => ()): T = {
+    val statsD = mock[StatsD]
+    SharedMetricRegistries.clear()
+    SharedHealthCheckRegistries.clear()
+    val reporter = StatsDReporter.forRegistry(SharedMetricRegistries.getOrCreate("default"))
+      .convertRatesTo(TimeUnit.SECONDS)
+      .convertDurationsTo(TimeUnit.MILLISECONDS)
+      .build(statsD)
+    reporter.start(1, TimeUnit.SECONDS)
+    try {
+      val result = testCode
+      eventually(timeout(10 seconds)) {
+        val order = mockitoInOrder(statsD)
```

Comment:
I haven't used Mockito before.  Let me see if I have interpreted this correctly.

* record the calls to `statsD` methods in order
* verify that the calls were made in the order `connect()`, `send()` at least once, and then `close()`
* enable callers to access the arguments given to `send()` for further test verification